frontend/src/hooks/useGetPipelineToolInfoEffect.ts

```diff
@@ -55,7 +57,7 @@ export const useGetPipelineToolInfoEffect = (): IUseVerifyPipeLineToolStateInter
   ]);
 
   useEffect(() => {
-    if (!apiTouchedRef.current && !isLoading) {
+    if (!apiTouchedRef.current && !isLoading && shouldRefresh) {
```

Comment:
React Hook useEffect has a missing dependency: 'shouldRefresh'. Either include it or remove the dependency array